tket2/src/optimiser/badger.rs

```diff
@@ -440,7 +441,7 @@ where
         logger.log_best(circ_cost.clone(), num_rewrites);
 
         let (joins, rx_work): (Vec<_>, Vec<_>) = chunks
-            .iter_mut()
```

Comment:
I wasn't sure if this would improve the runtime, since the `map` block below is mostly just about creating new threads. But a quick test showed that this whole `let (joins, rx_works) = ...` block is now a lot faster!

On my laptop, trying multiple shots of
```
cargo run -p badger-optimiser --release -- -e test_files/eccs/nam_6_3.rwr -i 
test_files/barenco_tof_10.json --split-circ -j N
```
while measuring the block time I get

| Threads (`N`) | Block time with `iter_mut` | with `par_iter_mut` | Improvement |
| ------------ | ------- | --------- | -------- |
| 2 | 100ms | 57ms | 0.57 |
| 10 | 719ms | 471ms | 0.65 |